ts/src/index.ts

```diff
@@ -90,22 +113,41 @@ export async function initConfig(config: Config): Promise<ProfilerConfig> {
                     // ignore errors, which will occur when not on GCE.
                 }) ||
         [undefined, undefined];
-    if (!mergedConfig.zone && zone) {
-      mergedConfig.zone = zone.substring(zone.lastIndexOf('/') + 1);
+    if (!config.zone && zone) {
+      config.zone = zone.substring(zone.lastIndexOf('/') + 1);
     }
-    if (!mergedConfig.instance && instance) {
-      mergedConfig.instance = instance;
+    if (!config.instance && instance) {
+      config.instance = instance;
     }
   }
+  return config;
+}
 
-  if (!hasService(mergedConfig)) {
-    throw new Error('Service must be specified in the configuration.');
+/**
+ * Initializes the config, and starts heap profiler if the heap profiler is
+ * needed. Returns a profiler if creation is successful. Otherwise, returns
+ * rejected promise.
+ */
+export async function createProfiler(config: Config): Promise<Profiler> {
+  if (!semver.satisfies(process.version, pjson.engines.node)) {
+    throw new Error(
+        `Could not start profiler: node version ${process.version}` +
+        ` does not satisfies "${pjson.engines.node}"`);
   }
 
-  return mergedConfig;
-}
+  let profilerConfig: ProfilerConfig;
```

Comment:
This and next line could be single `let profilerConfig = initConfigLocal(config);` line?

Reply:
Done.